src/main/groovy/org/pih/warehouse/importer/CycleCountItemsExcelImporter.groovy

```diff
@@ -0,0 +1,68 @@
+package org.pih.warehouse.importer
+
+import grails.util.Holders
+import org.grails.plugins.excelimport.AbstractExcelImporter
+import org.grails.plugins.excelimport.DefaultImportCellCollector
+import org.grails.plugins.excelimport.ExcelImportService
+import org.grails.plugins.excelimport.ExpectedPropertyType
+
+class CycleCountItemsExcelImporter extends AbstractExcelImporter implements DataImporter {
+
+    ExcelImportService excelImportService
+
+    static CELL_REPORTER = new DefaultImportCellCollector()
+
+    static Map COLUMN_MAP = [
+            sheet    : "Sheet1",
+            startRow : 1,
+            columnMap: [
+                    A: "cycleCountId",
+                    B: "productCode",
+                    C: "product.name",
+                    D: "lotNumber",
+                    E: "expirationDate",
+                    F: "binLocation",
+                    G: "quantityCounted",
+                    H: "comment",
+            ]
+    ]
+
+    static Map PROPERTY_MAP = [
+            cycleCountId:             ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
+            productCode:              ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
+            "product.name":           ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
+            "lotNumber":              ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
+            "expirationDate":         ([expectedType: ExpectedPropertyType.DateJavaType, defaultValue: null]),
```

Comment:
What happens if the date is poorly formatted at this point? Does the import fail? If so we should probably think of a different solution since we eventually want users to be able to resolve bad data in the UI, In a previous tech huddle we discussed potentially making all these fields StringType and letting the frontend handle all the validation. But if the import doesn't fail this is fine.

Reply:
I checked that, and if you e.g. provide a random string instead of a date, it is parsed to `null`, so I believe it's expected.

Reply:
This is fine for now but we might want to refactor this later to return it to the client as a string so that we can notify the user that they provided an invalid date. Otherwise if we just return null when we have an invalid format we wont be able to distinguish between the case where the lot just doesnt't have an expiry date (because it also returns null)